src/sortStudents.ts

```diff
@@ -1,16 +1,67 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calcAverage(grades: number[]): number {
+  const sumOfGrades: number = grades.reduce((sum: number, current: number) => (
+    sum + current
+  ));
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sumOfGrades / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  let studentsCopy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      studentsCopy = studentsCopy.sort((student1, student2) => {
+        return order === 'asc'
+          ? student1[sortBy].localeCompare(student2[sortBy])
+          : student2[sortBy].localeCompare(student1[sortBy]);
+      });
+      break;
+
```

Comment:
It's a good tone to use curly brackets inside case
```suggestion
    case SortType.Surname: {
      studentsCopy = studentsCopy.sort((student1, student2) => {
        return order === 'asc'
          ? student1[sortBy].localeCompare(student2[sortBy])
          : student2[sortBy].localeCompare(student1[sortBy]);
      });
      break;
      }
```

Reply:
fixed!